R/utils.R

```diff
@@ -157,17 +157,31 @@ with_rprofile_user <- function(new, code) {
 
 untar <- function(tarfile, ...) {
   if (os_type() == "windows") {
-    status <- try(utils::untar(tarfile, extras = "--force-local", ...))
-    if(inherits(status, "try-error") || status != 0){
+    tarhelp <- system2("tar", "--help", stdout = TRUE, stderr = TRUE)
```

Comment:
If `Sys.getenv("TAR")` is 'internal' we should probably just skip this completely. Also what does `system2()` do if there is no tar executable on the PATH?

OTOH `extras` is ignored if using the internal tar implementation IIRC, so maybe it is fine as is.

Reply:
Oh, yeah, `system2` needs to  be `tryCatch()`-ed. Thanks for catching this.